build/build.py

```diff
@@ -90,40 +90,40 @@ def check_numpy_version(python_bin_path):
 
 # Bazel
 
-BAZEL_BASE_URI = "https://github.com/bazelbuild/bazel/releases/download/4.1.0/"
+BAZEL_BASE_URI = "https://github.com/bazelbuild/bazel/releases/download/4.2.1/"
 BazelPackage = collections.namedtuple("BazelPackage",
                                       ["base_uri", "file", "sha256"])
 bazel_packages = {
     ("Linux", "x86_64"):
         BazelPackage(
```

Comment:
Github doesn't let me comment on unchanged lines.

Can you also update the version check on line 197? And other references of 4.1.0?

Reply:
I'm intentionally not doing that. Bazel itself is supposed to check the .bazelverison. The check on line 197 is mostly there to defend against a really old bazel version which doesn't have that check.

Reply:
Changed my mind and updated it since it's probably less confusing to users who hit the message this way.